src/kvasir/mpl/functions/arithmetic/decrement.hpp

```diff
@@ -7,6 +7,6 @@
 namespace kvasir {
 	namespace mpl {
 		template <typename A>
-		using decrement = integral_constant<decltype(A{} - 1), (A{} - 1)>;
+		using decrement = integral_constant<decltype(A::value - 1), (A::value - 1)>;
```

Comment:
I'm not sure integral_constant<short,5> is a good idea but wouldn't this return an integral_constant<int,4> because of integer promotion?